### DISCUSSION TITLE
How to detect that VS Code Extension is run in Theia IDE?

### DISCUSSION THREAD

**Discussion by MarshalX** [eclipse-theia/theia#14396] [Q&A] [answered]
Hi! I am VS Code extension dev. I want to understand in my codebase in which IDE extension was run. How I can do it? Will be it okay to check something like `process.env['THEIA_PARENT_PID']` or there is better solution?

**Comment by msujew** [comment, marked as answer]
Hey @MarshalX,

Generally we attempt to act as VS Code. However, I assume there are quite a few things that we accidentally expose. For example, the `vscode.env.appRoot` variable points to the current Theia instance. In there, files that are only used in VS Code (like the `product.json`) will be missing. As for the environment variables: I don't assume that any env variables that we push into the extension host are subject to change, so you're likely fine with depending on them.

There is no real API for checking whether the app runs in VS Code or Theia.

**Reply by MarshalX** [reply]
Thank you @msujew! The reason why I ask is because in some cases Theia acts differently or does not acts at all as VS Code API. That's why I want to separate some logic because proper fixes in Theia could be time consuming. For example command like `workbench.action.problems.focus` do not work. Tree view do not collapse. I though to disable some actions for Theia as quick win for us. I'll try `vscode.env.appRoot`, thank you!

**Reply by tsmaeder** [reply]
Hi @MarshalX: if you find things in the extension API that are not working like in VS Code, we'd really appreciate it if you open an issue.
